vault_core/src/vault.rs

```diff
@@ -539,6 +606,48 @@ impl Vault {
         Ok(())
     }
 
+    /// Checks if a withdrawal is allowed based on the current vault limits.
+    ///
+    /// If both `epoch_snapshot_amount` and `last_full_state_update_slot` are zero, this indicates that the vault has not yet recorded any withdrawals or state updates for the current epoch.
+    /// In this scenario, any withdrawal amount up to the total `max_withdrawable` tokens is allowed because the vault is effectively in an initial state.
+    ///
+    /// # Returns
+    /// * `Result<(), VaultError>` - Returns `Ok(())` if the withdrawal is within the allowed limit.
+    ///
+    /// # Errors
+    /// This function can return the following errors:
+    /// * [`jito_vault_sdk::error::VaultError::VaultOverflow`] - If any arithmetic operation (addition or multiplication) results in an overflow.
+    /// * [`jito_vault_sdk::error::VaultError::VaultWithdrawalLimitExceeded`] - If the requested withdrawal exceeds the allowed limit for the epoch.
+    pub fn check_withdrawal_allowed(&self, amount_to_withdraw: u64) -> Result<(), VaultError> {
+        let epoch_snapshot_amount: u64 = self.epoch_snapshot_amount();
+        let last_full_state_update_slot: u64 = self.last_full_state_update_slot();
+        let max_withdrawable = self.tokens_deposited();
+
+        if epoch_snapshot_amount == 0
+            && last_full_state_update_slot == 0
+            && amount_to_withdraw <= max_withdrawable
+        {
+            Ok(())
+        } else {
+            let total_withdraw_amount = self
+                .epoch_withdraw_amount()
+                .checked_add(amount_to_withdraw)
+                .ok_or(VaultError::VaultOverflow)?;
+            let max_allowed_withdraw = self
+                .epoch_snapshot_amount()
+                .checked_mul(self.epoch_withdraw_cap_bps() as u64)
+                .ok_or(VaultError::VaultOverflow)?
+                .div_ceil(10_000);
```

Comment:
nit: use constant for this now

Reply:
Other functions use `MAX_FEE_BPS`, but in this case, this constant does not match, I think.
So I created new constant `MAX_EPOCH_WITHDRAW_BPS`, though both values are same.